### PR TITLE
fix: Styling fixes for horizontal filter bar

### PR DESCRIPTION
**Pull request by kgabryje**
### SUMMARY
A few styling fixes related to horizontal native filters bar (and 1 for vertical):
- Remove extra right padding from empty state text
- Remove extra spacing to right of Add/Edit Filters button
- Make Apply and Clear All buttons aligned with each other
- Increase padding at end of non-overflowed items
- Increase top/bottom padding of horizontal bar
- Align Apply and Clear All buttons with other Dashboard buttons

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

CC @kasiazjc @codyml for review

### CODE REVIEW

**Comment by kgabryje** [comment]
@michael-s-molina Thank you for suggestions, applied both changes.
@kasiazjc also agrees

image

**Comment by michael-s-molina** [comment]
image

@kgabryje Final nit: Should the distance between the gear icon, Add Filter, and empty text be the same?

**Comment by kgabryje** [comment]
@michael-s-molina Spacings between the COMPONENTS are the same - 16px. The problem is that the icon component has some empty space around it. When we start using the icons from the new design system, the problem will be solved, since those new icons don't have spacing around them

image

**Review by michael-s-molina — APPROVED** [review]
LGTM. Thank you for the fixes and for addressing the comments @kgabryje.

File: superset-frontend/src/dashboard/components/nativeFilters/FilterBar/Header/index.tsx [hunk 1]
```diff
@@ -35,29 +35,40 @@ import { RootState } from 'src/dashboard/types';
...
-  .anticon {
-    padding-top: ${({ theme }) => `${theme.gridUnit + 2}px`};
-  }
 `;
 
 const Wrapper = styled.div`
   ${({ theme }) => `
-    padding: ${theme.gridUnit}px ${theme.gridUnit * 2}px;
+    padding: ${theme.gridUnit * 3}px ${theme.gridUnit * 2}px ${
+    theme.gridUnit
```
Comment (rusackas) [inline superset-frontend/src/dashboard/components/nativeFilters/FilterBar/Header/index.tsx:70]: that's some goofy formatting, but if the linter's happy, I'm happy.

**Review by rusackas — APPROVED** [review]
Code LGTM! Will try to remember to spin up a test env when CI is ready for that.

**Review by geido — APPROVED** [review]

**Pull request merged**